### PR TITLE
Improvements to Tril-based compiler tests

### PR DESCRIPTION
**Pull request by Leonardo2718**
This change set makes some general improvements to the Tril-based compiler tests.

### CODE REVIEW

File: fvtest/compilertriltest/JitTest.hpp [hunk 1]
```diff
@@ -40,7 +42,8 @@ class JitTest : public ::testing::Test
 
    static void SetUpTestCase()
       {
-      ASSERT_TRUE(initializeJit()) << "Failed to initialize the JIT.";
+      auto initSuccess = initializeJitWithOptions((char*)"-Xjit:acceptHugeMethods,enableBasicBlockHoisting,omitFramePointer,useIlValidator,paranoidoptcheck");
+      ASSERT_TRUE(initSuccess) << "Failed to initialize the JIT.";
```
Comment (mgaudet) [inline fvtest/compilertriltest/JitTest.hpp]: Do we _need_ `omitFramePointer`, or is this just to match the JitBuilder init code?

Comment (Leonardo2718) [inline fvtest/compilertriltest/JitTest.hpp]: Yes, it's just there to match JitBuilder's init code.

Until we have a backend that provides more control to test writers, I think it's best to keep things consistent.

File: fvtest/compilertriltest/JitTest.hpp [hunk 2]
```diff
@@ -49,4 +57,68 @@ class JitTest : public ::testing::Test
       }
    };
 
+/**
+ * @brief Returns the Cartesian product of two standard-conforming containers
+ * @tparam L is the type of the first input standard-conforming container
+ * @tparam R is the type of the second input standard-conforming container
+ * @param l is the first input standard-conforming container
+ * @param r is the second input standard-conforming container
+ * @return the cartesion product of the input containers as a std::vector of std::tuple
```
Comment (0xdaryl) [inline fvtest/compilertriltest/JitTest.hpp]: typo:  Cartesian

File: fvtest/compilertriltest/JitTest.hpp [hunk 3]
```diff
@@ -49,4 +57,68 @@ class JitTest : public ::testing::Test
...
+
+/**
+ * @brief Returns the Cartesian product of two initializer lists
+ * @tparam L is the type of elements in the first initializer list
+ * @tparam R is the type of elements in the second initializer list
+ * @param l is the first initializer list
+ * @param r is the second initializer list
+ * @return the Cartesian product of the two input lists
+ *
+ * Becuase of the rules surrounding type-deduction of initializer lists, the
```
Comment (0xdaryl) [inline fvtest/compilertriltest/JitTest.hpp]: Typo:  Because

File: fvtest/compilertriltest/JitTest.hpp [hunk 4]
```diff
@@ -49,4 +57,68 @@ class JitTest : public ::testing::Test
...
+ * types of the elements of the two lists must be explicitely specified when
```
Comment (0xdaryl) [inline fvtest/compilertriltest/JitTest.hpp]: Typo: explicitly

File: fvtest/compilertriltest/JitTest.hpp (hunk 2 above)
Comment (Leonardo2718) [inline fvtest/compilertriltest/JitTest.hpp]: Fixed.

File: fvtest/compilertriltest/JitTest.hpp (hunk 3 above)
Comment (Leonardo2718) [inline fvtest/compilertriltest/JitTest.hpp]: Fixed.

**Pull request merged**